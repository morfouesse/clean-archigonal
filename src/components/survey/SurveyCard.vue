<template>
  <v-card elevation="3" class="mx-auto" width="300">
    <template v-slot:title>
      <span class="title">{{ survey.label }}</span>
    </template>
    <v-list>
      <v-list-group>
        <template v-slot:activator="{ props }">
          <v-list-item v-bind="props" class="title" title="Voir les questions"></v-list-item>
        </template>

        <v-list-item
          v-for="(question, questionIndex) in survey.questions"
          :key="questionIndex"
          :title="question.label"
        >
          <v-list-group>
            <template v-slot:activator="{ props }">
              <v-list-item v-bind="props" class="title" title="Voir les réponses"></v-list-item>
            </template>
            <v-list-item
              v-for="(answer, answerIndex) in question.answers"
              :key="answerIndex"
              :title="answer.label"
            ></v-list-item>
          </v-list-group>
        </v-list-item>
      </v-list-group>
    </v-list>
    <v-card-actions>
      <v-btn variant="tonal">joué</v-btn>
    </v-card-actions>
  </v-card>
</template>
<script setup lang="ts">
import type { SurveyViewModel } from '@/domains/survey/ports/fetchSurvey.presenter'

const { survey } = defineProps<{
  survey: SurveyViewModel
}>()
</script>
