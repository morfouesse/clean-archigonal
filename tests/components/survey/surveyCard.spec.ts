import SurveyCard from '@/components/survey/SurveyCard.vue'
import type { SurveyViewModel } from '@/domains/survey/ports/fetchSurvey.presenter'
import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'

describe('Survey card', () => {
  // https://dev.to/jacobandrewsky/testing-vue-components-with-vitest-5c21
  it('snapshot', () => {
    const survey: SurveyViewModel = {
      label: 'les tests',
      questions: [
        {
          label: 'Faut-il faire des tests ?',
          answers: [
            {
              label: 'OUIIIII',
            },
            {
              label: 'heuuu, connait pas',
            },
          ],
        },
        {
          label: 'Quelles tests en priorit√© ?',
          answers: [
            {
              label: 'les tests unitaires',
            },
            {
              label: 'les tests E2E',
            },
          ],
        },
      ],
    }

    const wrapper = mount(SurveyCard, {
      props: {
        survey: survey,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
