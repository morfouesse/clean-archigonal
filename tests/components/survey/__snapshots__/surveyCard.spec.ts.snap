// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Survey card > snapshot 1`] = `
"<div class="v-card v-theme--light v-card--density-default elevation-3 v-card--variant-elevated mx-auto" style="width: 300px;">
  <!---->
  <div class="v-card__loader">
    <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
      <!---->
      <div class="v-progress-linear__background"></div>
      <div class="v-progress-linear__buffer" style="width: 0%;"></div>
      <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
        <div class="v-progress-linear__indeterminate">
          <div class="v-progress-linear__indeterminate long"></div>
          <div class="v-progress-linear__indeterminate short"></div>
        </div>
      </transition-stub>
      <!---->
    </div>
  </div>
  <div class="v-card-item">
    <!---->
    <div class="v-card-item__content">
      <div class="v-card-title"><span class="title">les tests</span></div>
      <!---->
      <!---->
    </div>
    <!---->
  </div>
  <!---->
  <div class="v-list v-theme--light v-list--density-default v-list--one-line" tabindex="0" role="listbox">
    <div class="v-list-group">
      <div class="v-list-item v-list-item--link v-theme--light v-list-item--density-default v-list-item--one-line rounded-0 v-list-item--variant-text v-list-group__header title" tabindex="-2" id="v-list-group--id-Symbol(nested item)"><span class="v-list-item__overlay"></span><span class="v-list-item__underlay"></span>
        <!---->
        <div class="v-list-item__content" data-no-activator="">
          <div class="v-list-item-title">Voir les questions</div>
          <!---->
          <!---->
        </div>
        <div class="v-list-item__append"><i class="mdi-chevron-down mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true" density="default"></i>
          <!---->
          <div class="v-list-item__spacer"></div>
        </div>
      </div>
      <transition-stub name="" appear="false" persisted="false" css="false">
        <div class="v-list-group__items" role="group" aria-labelledby="v-list-group--id-Symbol(nested item)" style="display: none;">
          <div class="v-list-item v-theme--light v-list-item--density-default v-list-item--one-line rounded-0 v-list-item--variant-text">
            <!----><span class="v-list-item__underlay"></span>
            <!---->
            <div class="v-list-item__content" data-no-activator="">
              <div class="v-list-item-title">Faut-il faire des tests ?</div>
              <!---->
              <div class="v-list-group">
                <div class="v-list-item v-list-item--link v-theme--light v-list-item--density-default v-list-item--one-line rounded-0 v-list-item--variant-text v-list-group__header title" tabindex="-2" id="v-list-group--id-Symbol(nested item)"><span class="v-list-item__overlay"></span><span class="v-list-item__underlay"></span>
                  <!---->
                  <div class="v-list-item__content" data-no-activator="">
                    <div class="v-list-item-title">Voir les réponses</div>
                    <!---->
                    <!---->
                  </div>
                  <div class="v-list-item__append"><i class="mdi-chevron-down mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true" density="default"></i>
                    <!---->
                    <div class="v-list-item__spacer"></div>
                  </div>
                </div>
                <transition-stub name="" appear="false" persisted="false" css="false">
                  <div class="v-list-group__items" role="group" aria-labelledby="v-list-group--id-Symbol(nested item)" style="display: none;">
                    <div class="v-list-item v-theme--light v-list-item--density-default v-list-item--one-line rounded-0 v-list-item--variant-text">
                      <!----><span class="v-list-item__underlay"></span>
                      <!---->
                      <div class="v-list-item__content" data-no-activator="">
                        <div class="v-list-item-title">OUIIIII</div>
                        <!---->
                        <!---->
                      </div>
                      <!---->
                    </div>
                    <div class="v-list-item v-theme--light v-list-item--density-default v-list-item--one-line rounded-0 v-list-item--variant-text">
                      <!----><span class="v-list-item__underlay"></span>
                      <!---->
                      <div class="v-list-item__content" data-no-activator="">
                        <div class="v-list-item-title">heuuu, connait pas</div>
                        <!---->
                        <!---->
                      </div>
                      <!---->
                    </div>
                  </div>
                </transition-stub>
              </div>
            </div>
            <!---->
          </div>
          <div class="v-list-item v-theme--light v-list-item--density-default v-list-item--one-line rounded-0 v-list-item--variant-text">
            <!----><span class="v-list-item__underlay"></span>
            <!---->
            <div class="v-list-item__content" data-no-activator="">
              <div class="v-list-item-title">Quelles tests en priorité ?</div>
              <!---->
              <div class="v-list-group">
                <div class="v-list-item v-list-item--link v-theme--light v-list-item--density-default v-list-item--one-line rounded-0 v-list-item--variant-text v-list-group__header title" tabindex="-2" id="v-list-group--id-Symbol(nested item)"><span class="v-list-item__overlay"></span><span class="v-list-item__underlay"></span>
                  <!---->
                  <div class="v-list-item__content" data-no-activator="">
                    <div class="v-list-item-title">Voir les réponses</div>
                    <!---->
                    <!---->
                  </div>
                  <div class="v-list-item__append"><i class="mdi-chevron-down mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true" density="default"></i>
                    <!---->
                    <div class="v-list-item__spacer"></div>
                  </div>
                </div>
                <transition-stub name="" appear="false" persisted="false" css="false">
                  <div class="v-list-group__items" role="group" aria-labelledby="v-list-group--id-Symbol(nested item)" style="display: none;">
                    <div class="v-list-item v-theme--light v-list-item--density-default v-list-item--one-line rounded-0 v-list-item--variant-text">
                      <!----><span class="v-list-item__underlay"></span>
                      <!---->
                      <div class="v-list-item__content" data-no-activator="">
                        <div class="v-list-item-title">les tests unitaires</div>
                        <!---->
                        <!---->
                      </div>
                      <!---->
                    </div>
                    <div class="v-list-item v-theme--light v-list-item--density-default v-list-item--one-line rounded-0 v-list-item--variant-text">
                      <!----><span class="v-list-item__underlay"></span>
                      <!---->
                      <div class="v-list-item__content" data-no-activator="">
                        <div class="v-list-item-title">les tests E2E</div>
                        <!---->
                        <!---->
                      </div>
                      <!---->
                    </div>
                  </div>
                </transition-stub>
              </div>
            </div>
            <!---->
          </div>
        </div>
      </transition-stub>
    </div>
  </div>
  <div class="v-card-actions"><button type="button" class="v-btn v-btn--slim v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-tonal"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
      <!----><span class="v-btn__content" data-no-activator="">joué</span>
      <!---->
      <!---->
    </button></div>
  <!---->
  <!----><span class="v-card__underlay"></span>
</div>"
`;
